src/components/accession2/withdraw/WithdrawModal.tsx

```diff
@@ -464,49 +518,113 @@ export default function WithdrawDialog(props: WithdrawDialogProps): JSX.Element
             </Grid>
           </>
         ) : null}
-        <Grid item xs={12} paddingBottom={2}>
-          <Box display='flex' alignItems={fieldsErrors.withdrawnQuantity ? 'center' : 'end'}>
-            <Textfield
-              label={strings
-                .formatString(
-                  strings.AMOUNT_REMAINING,
-                  `${accession.remainingQuantity?.quantity} ${accession.remainingQuantity?.units}`
-                )
-                .toString()}
-              id='withdrawnQuantity'
-              onChange={(value) => onChangeWithdrawnQuantity(Number(value))}
-              type='text'
-              value={
-                isNurseryTransfer
-                  ? nurseryTransferRecord.germinatingQuantity
-                  : record.withdrawnQuantity?.quantity.toString()
-              }
-              errorText={fieldsErrors.withdrawnQuantity}
-            />
-            <Box paddingLeft={1}>
-              {accession.remainingQuantity?.units === 'Seeds' ||
-              record.purpose === 'Viability Testing' ||
-              isNurseryTransfer ? (
-                <Box>{strings.CT}</Box>
-              ) : (
-                <Dropdown
-                  options={preferredUnits}
-                  placeholder={strings.SELECT}
-                  onChange={onChangeUnit}
-                  selectedValue={record.withdrawnQuantity?.units}
-                  fullWidth={true}
+        <Box className={classes.withdraw}>
```

Comment:
Here again we may benefit from having flavors of components based on semantics. It may be a bit tricky but helps avoid the various if/else logic. The common stuff could be a separate shared component. This is mostly a theoretical recommendation.

Example:
```
Withdraw Modal
  .. all the withdraw stuff
  - withdraw by selector
  - ByWeightComponent (if selected by weight)
     - weight specific log
     - use a shared sub component for the input/withdraw-all elements
  - ByQuantityComponent (if selected by quantity)
     - quantity specific logic
     - use a shared sub component for the input/withdraw-all elements
  ... rest of the withdraw stuff
```
Here again there might be some properties passing to be done.
Again, my suggestion is mostly theoretical to help separate context and less centralized if/else logic. It may end up being a waste of time on its own but maybe worth trying if we think this component is a tad complex.

Reply:
I ended up separating out the components, but didn't try to share too much logic between the two.  The simplification in the logic was well worth it.  Would be interested to hear if you think trying to re-combine some functionality would be a worthwhile exercise.

Reply:
This is so much cleaner and manageable. I don't see anything worth re-combining, they are very different. Withdraw-all is a common feature but underlying logic is different enough that a refactor won't really help.